mungegithub/mungers/release-note-label.go

```diff
@@ -144,20 +144,15 @@ func (r *ReleaseNoteLabel) Munge(obj *github.MungeObject) {
 		return
 	}
 
+	releaseNoteLabelNeededAdded := false
 	if !obj.HasLabel(releaseNoteLabelNeeded) {
 		obj.AddLabel(releaseNoteLabelNeeded)
+		releaseNoteLabelNeededAdded = true
 	}
 
-	if !obj.HasLabel(lgtmLabel) {
-		return
+	if releaseNoteLabelNeededAdded && obj.HasLabel(lgtmLabel) {
+		obj.WriteComment(releaseNoteBody)
 	}
-
-	if obj.HasLabel(doNotMergeLabel) {
-		return
-	}
-
-	obj.WriteComment(releaseNoteBody)
-	obj.AddLabel(doNotMergeLabel)
```

Comment:
And what if PR doesn't have `do-not-merge` and we are no longer setting it there.
In that case in 1st call of `Muge` we will add `release-note-label-needed` and comment on PR (no more adding `do-not-merge`, but then:
2nd call (please follow Munge function).
It will check if object have one of 4 release note labels. It doesn't.
It will check if `release-note-label-needed` is present - it is.
Then it will hit `if obj.HasLabel(doNotMergeLabel)` but now we didn't added that label in 1st call so it will proceed and add comment agaig.
The same in 3rd call, 4th call etc.
Each `Munge` call will add `obj.WriteComment(releaseNoteBody`

Before removing single line `obj.AddLabel(doNotMergeLabel)` it was adding `do-not-merge` label in 1st call, so any next call will check `if obj.HasLabel(doNotMergeLabel) {` and then return - so it won't add comment more than once.

So I think my changes are actually OK.
Because I'm adding additional condition to put comment only in the same call when `release-note-lable-needed` were added.

If I'm wrong please let me know - but please describe why I'm wrong, this is my 2nd PR, and maybe I missed something...
But please see this file:
https://github.com/kubernetes/test-infra/blob/master/mungegithub/mungers/release-note-label.go
And `Munge` function.
Please remove only one line (as You wanted me to do) remove: `obj.AddLabel(doNotMergeLabel)` (last line)
And then imagine what will happen if we call `Munge` 5 times. Comment `obj.WriteComment(releaseNoteBody)` would be added 5 times then.
Then please take a look on my implementation in this PR.
My implementation will not add `do-not-merge` label but also wont' add `obj.WriteComment(releaseNoteBody)` more than once.

If I'm wrong please correct me.... what exactly I'm assuming bad.
Maybe `Munge` function is called only in specific conditions? Myabe this is what I missed?
BTW: when `Munge` is called? (for this particular munger)?
I know that `prow` reacts to Github web hooks.... how about mungegithub? Periodic?